### PR TITLE
Allow load ARM9 payload from memory

### PR DESCRIPTION
**Pull request by d0k3**
Useful for when Brahma is used as library

### CODE REVIEW

File: source/brahma.c [hunk 1]
```diff
@@ -283,6 +283,22 @@ s32 load_arm9_payload (char *filename) {
...
+   data: array of u8 containing the payload
+   dsize: size of the data array
+   returns: 0 on failure, 1 on success */
+s32 load_arm9_payload_from_mem (u8* data, u32 dsize) {
+	s32 result = 0;
+
+	if (dsize >= 8 && (dsize <= ARM9_PAYLOAD_MAX_SIZE)) {
+		g_ext_arm9_size = dsize;
+		memcpy(g_ext_arm9_buf, data, dsize);
+		result = g_ext_arm9_loaded = 1;
```
Comment (archshift) [inline source/brahma.c:296]: `result` is written-to but never read.

Comment (patois) [inline source/brahma.c:296]: also, `data` should be added sanity checking

**Comment by patois** [comment]
please squash your commits into a single one

**Comment by d0k3** [comment]
Rebased!

**Pull request merged**